xarray/core/concat.py

```diff
@@ -665,10 +678,6 @@ def get_indexes(name):
         else:
             index_vars = index.create_variables()
         result[dim] = index_vars[dim]
-        result_indexes[dim] = index
-
-    # TODO: add indexes at Dataset creation (when it is supported)
-    result = result._overwrite_indexes(result_indexes)
```

Comment:
Removing those lines don't break any existing test because the line above `result[dim] = index_vars[dim]` actually re-creates a default PandasIndex when assigning the new `dim` variable. However, this unnecessarily re-creates a new index (or re-wrap an existing one) and this may not work in the future if we allow passing a custom xarray index as `dim` argument to `concat`.

It would be better to explicitly add *both* `index` and `index_vars` to `result`. Best way would be to assign them to `result_indexes` and `coord_vars` respectively before constructing the `Coordinates` object and then the `result` object, unless there are cases where `result.drop_vars(unlabeled_dims)` would delete the index coordinate.

Reply:
I think I've addressed your comment now @benbovy